ethcore/src/client/client.rs

```diff
@@ -178,15 +178,15 @@ impl Client {
 		db_config.compaction = config.db_compaction.compaction_profile();
 		db_config.wal = config.db_wal;
 
-		let db = Arc::new(Database::open(&db_config, &path.to_str().unwrap()).expect("Error opening database"));
+		let db = Arc::new(try!(Database::open(&db_config, &path.to_str().unwrap()).map_err(|s| ClientError::Database(s))));
```

Comment:
`s` is a `String`, not a `DatabaseError` and implementing `ClientError::From<String>` is a little to general.